File: Codeforces/Div3/786/Infinite-Replacement/main.cpp
// https://codeforces.com/contest/1674/problem/C

#include<bits/stdc++.h>

using namespace std;

long long int checkCombinations(string s, string t) {
    int m = s.size();
    long long int answ = 1;

    if (t.size() == 1 && t[0] == 'a')
        return 1L;
        
    for (auto a : t)
        if (a == 'a')
            return -1;

    for (int i=0; i<m; i++)
        answ <<= 1;

    return answ;
}

int main () {
    cin.tie(0);
    ios_base::sync_with_stdio(0);

    int t; cin >> t;

    while(t--) {
        string s, t;
        cin >> s >> t;
        cout << checkCombinations(s, t) << endl;
    }

    return 0;
}

File: Codeforces/Div3/786/Infinite-Replacement/main.kt
// https://codeforces.com/contest/1674/problem/C

fun readLn() = readLine()!!
fun readInt() = readLn().toInt()

fun checkCombinations (s : String, t : String) : Long {
    var answ : Long = 1

    if (t.length.equals(1) && t.equals("a"))
        return 1L

    for (a : Char in t)
        if (a.equals('a'))
            return -1L

    for (i in 1..s.length)
        // Shift left bits
        answ = answ shl 1

    return answ
}

fun main () {
    for (i in 1..readInt())
        println(checkCombinations(readLn(), readLn()))
}
